lib/tree.rb

```diff
@@ -104,22 +124,48 @@ def self.find_tree(repository, file_path, revstring=Constants::HEAD)
 
     private
 
-    def jtree_entries(options={})
+    def contents_array
+      @contents_ary ||= jtree_entries
```

Comment:
Note: we are storing the contents of the Tree in `@contents_array` (and similarly for `recursive_contents_array`). This is efficient if we expect these methods to be called multiple times on the same objects, but inefficient (in terms of memory and garbage collection) if we expect it to typically be called only once. I'm not sure if we're betting on the right horse currently. And of course, the user can always store the result themselves if they're planning to reuse it multiple times (e.g. `myvar = tree.each.to_a`).

Reply:
Fair point. 👍 I'm not sure either, but I'm fine with dropping the internal caching here.

Reply:
I'll keep it for now, so something intuitive like `tree = ...; tree.trees; tree.count` will not trigger two loops. But we should keep in mind this possibly area of object proliferation. :)